pkg/instancegroups/instancegroups.go

```diff
@@ -221,6 +236,80 @@ func (r *RollingUpdateInstanceGroup) RollingUpdate(rollingUpdateData *RollingUpd
 	return nil
 }
 
+func (r *RollingUpdateInstanceGroup) maybeTaintAllNeedUpdate(update []*cloudinstances.CloudInstanceGroupMember, rollingUpdateData *RollingUpdateCluster, noneReady bool, uIdx int) error {
+	if r.CloudGroup.InstanceGroup.Spec.Role != api.InstanceGroupRoleNode || rollingUpdateData.CloudOnly {
+		return nil
+	}
+
+	if noneReady {
+		// Wait until after one node is deleted and its replacement validates before the mass-cordoning
+		// in case the current spec does not result in usable nodes.
+		if uIdx != 1 || len(update) < 2 {
+			return nil
+		}
+	} else {
+		if uIdx != 0 || len(update) < 1 {
+			return nil
+		}
+	}
+
+	var toTaint []*corev1.Node
+	for _, u := range update {
+		if u.Node != nil && !u.Node.Spec.Unschedulable {
+			foundTaint := false
+			for _, taint := range u.Node.Spec.Taints {
+				if taint.Key == rollingUpdateTaintKey {
+					foundTaint = true
+				}
+			}
+			if !foundTaint {
+				toTaint = append(toTaint, u.Node)
+			}
+		}
+	}
+	if len(toTaint) > 0 {
+		noun := "nodes"
+		if len(toTaint) == 1 {
+			noun = "node"
+		}
+		klog.Infof("Tainting %d %s in %q instancegroup.", len(toTaint), noun, r.CloudGroup.InstanceGroup.Name)
+		for _, n := range toTaint {
+			if err := r.patchTaint(rollingUpdateData, n); err != nil {
+				if rollingUpdateData.FailOnDrainError {
+					return fmt.Errorf("failed to taint node %q: %v", n, err)
+				}
+				klog.Infof("Ignoring error tainting node %q: %v", n, err)
+			}
+		}
+	}
+	return nil
+}
+
+func (r *RollingUpdateInstanceGroup) patchTaint(rollingUpdateData *RollingUpdateCluster, node *corev1.Node) error {
+	oldData, err := json.Marshal(node)
+	if err != nil {
+		return err
+	}
+
+	node.Spec.Taints = append(node.Spec.Taints, corev1.Taint{
+		Key:    rollingUpdateTaintKey,
+		Effect: corev1.TaintEffectNoSchedule,
+	})
+
+	newData, err := json.Marshal(node)
+	if err != nil {
+		return err
+	}
+
+	patchBytes, err := strategicpatch.CreateTwoWayMergePatch(oldData, newData, node)
+	if err != nil {
+		return err
+	}
+
+	_, err = rollingUpdateData.K8sClient.CoreV1().Nodes().Patch(node.Name, types.StrategicMergePatchType, patchBytes)
```

Comment:
I was going to say it would be good to log the patch so we knew CreateTwoWayMergePatch was doing what we thought, but then I saw you have a test - even better 👍